src/util/file.rs

```diff
@@ -0,0 +1,77 @@
+use std::{
+    ffi::OsStr,
+    path::{Path, PathBuf},
+};
+
+pub async fn day_path<P: AsRef<Path>>(
+    root: P,
+    day: u32,
+) -> Result<std::path::PathBuf, std::io::Error>
+{
+    use std::{collections::VecDeque, io::*};
+    let dir_name = format!("day_{:02}", day);
+    let dir_name = OsStr::new(&dir_name);
+    let ignore = [OsStr::new("target"), OsStr::new(".git")];
+
+    let mut vec = VecDeque::new();
+    vec.push_back(root.as_ref().as_os_str().to_os_string());
+
+    while let Some(path) = vec.pop_front()
+    {
+        let mut stream = tokio::fs::read_dir(&path).await?;
+        while let Ok(Some(entry)) = stream.next_entry().await
+        {
+            let file_name = entry.file_name();
+            if ignore.contains(&file_name.as_os_str())
+            {
+                continue;
+            }
+
+            if file_name == dir_name
+            {
+                let mut buff: PathBuf = path.into();
+                buff.push(dir_name);
+                return Ok(buff);
+            }
+
+            let file_type = entry.file_type().await?;
+            if file_type.is_dir()
+            {
+                let mut path = Path::new(&path).to_path_buf();
+                path.push(entry.file_name());
+                let name = path.as_os_str().to_os_string();
+
+                vec.push_back(name);
+            }
+        }
+    }
+    Err(Error::new(ErrorKind::NotFound, "could not find target folder"))
```

Comment:
Nitpick, but should we have a different error message here? :thinking: Just wondering if it can be confusing with not finding the target folder when there also is a folder called target generated by cargo on compile

Reply:
fixed